log.c

```diff
@@ -140,11 +140,37 @@ void log_counter_info(void)
 void log_FTL_version(const bool crashreport)
 {
 	logg("FTL branch: %s", GIT_BRANCH);
-	logg("FTL version: %s", GIT_TAG);
+	logg("FTL version: %s", get_FTL_version());
 	logg("FTL commit: %s", GIT_HASH);
 	logg("FTL date: %s", GIT_DATE);
 	if(crashreport)
 		logg("FTL user: started as %s, ended as %s", username, getUserName());
 	else
 		logg("FTL user: %s", username);
 }
+
+static char *FTLversion = NULL;
+const char __attribute__ ((malloc)) *get_FTL_version(void)
+{
+	// Obtain FTL version if not already determined
+	if(FTLversion == NULL)
+	{
+		const char *tag = GIT_TAG;
+		if(strlen(tag) > 1)
+		{
+			FTLversion = strdup(GIT_VERSION);
+		}
+		else
+		{
+			const char *commit = GIT_HASH;
+			FTLversion = calloc(13, sizeof(char));
+			// Build version by appending 7 characters of the hash to "vDev-"
+			memcpy(FTLversion, "vDev-", 5);
+			memcpy(FTLversion+5, commit, 7);
+			// Zero-terminate string
+			FTLversion[12] = 0;
```

Comment:
This can be done in a simpler fashion:
```c
FTLversion = calloc(13, sizeof(char));
snprintf(FTLversion, 13, "vDev-%s", GIT_HASH);
```

Reply:
I used `memcpy` to prevent
```
log.c: In function ‘get_FTL_version’:
log.c:166:35: error: ‘%s’ directive output truncated writing 13 bytes into a region of size 8 [-Werror=format-truncation=]
  166 |    snprintf(FTLversion, 13, "vDev-%s", GIT_HASH);
      |                                   ^~
In file included from /usr/include/stdio.h:867,
                 from FTL.h:15,
                 from log.c:11:
/usr/include/bits/stdio2.h:67:10: note: ‘__builtin___snprintf_chk’ output 19 bytes into a destination of size 13
   67 |   return __builtin___snprintf_chk (__s, __n, __USE_FORTIFY_LEVEL - 1,
      |          ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
   68 |        __bos (__s), __fmt, __va_arg_pack ());
      |        ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

```
However, I now see that this can be avoided by using `%.7s` to explicitly specify that we want only the first seven bytes of `GIT_HASH`.